### PR TITLE
refactor addressit sanitizer for readability

### PR DESCRIPTION
**Pull request by missinglink**
this file was super hard to read so I had a crack at making it legible

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
   },
   "dependencies": {
     "@mapbox/geojson-extent": "^0.3.1",
-    "addressit": "1.5.0",
+    "addressit": "1.6.0",
```
Comment (orangejulius) [inline package.json]: lets handle this separately in https://github.com/pelias/api/pull/1122

**Comment by missinglink** [comment]
Huh very weird, I was expecting it to be a no-op refactor, I'll have a look at those tomorrow.

**Comment by orangejulius** [comment]
I told you, this wild old code will not go easily :P

**Comment by missinglink** [comment]
Actually those failures look like the exact matching PR, are you sure the didn't get mixed up?

**Comment by orangejulius** [comment]
I thought that too, and double checked. I will triple check in a minute

**Comment by missinglink** [comment]
wow ok, so this refactor was a useful experiment to remember how all the autocomplete functionality works.

I wrote up some docs explaining the autocomplete logic here: https://github.com/pelias/api/wiki/Autocomplete-search-logic

**Comment by missinglink** [comment]
This is fixed, it should now be a no-op refactor, which cleans it up significantly

File: sanitizer/_text_addressit.js [hunk 2]
```diff
@@ -1,10 +1,25 @@
-var check = require('check-types');
```
Comment (orangejulius) [inline sanitizer/_text_addressit.js]: yuss. I think now that we know and love lodash, we should eventually get rid of `check`. We are almost there

File: sanitizer/_text_addressit.js [hunk 3]
```diff
@@ -14,106 +29,84 @@ function _sanitize( raw, clean ){
...
-
-function parse(query, clean) {
-  var getAdminPartsBySplittingOnDelim = function(queryParts) {
-    // naive approach - for admin matching during query time
-    // split 'flatiron, new york, ny' into 'flatiron' and 'new york, ny'
-
-    var address = {};
+// naive approach - for admin matching during query time
+// split 'flatiron, new york, ny' into 'flatiron' and 'new york, ny'
+var naïve = function(tokens) {
```
Comment (orangejulius) [inline sanitizer/_text_addressit.js]: image

**Comment by missinglink** [comment]
Just a note on this, I was intending it to be a `no-op` but in the end, it had one minor change to the logic.

The `admin_parts` object was super confusing to deal with since it could be an `object`, it could be `undefined` and also in the case of only matching `admin_parts` it would be set to `null` :man_shrugging: 

So I've made that much easier to use by ensuring it always returns an object, and you just check if the object is empty or not.

There was one part of the code which did an `hasOwnProperty` check on the `null` which was previously true and now is false.

The effect of the change is that for addressit parses which only returned `admin_parts` (the AUS/USA state abbreviations) we used to consider the tokens as complete, we now consider them incomplete.

Good to merge.

**Comment by missinglink** [comment]
c8cw76gmohy11

**Pull request merged**